src/snapshot/src/lib.rs

```diff
@@ -394,11 +522,56 @@ mod tests {
         snapshot = Snapshot::load(&mut snapshot_mem.as_slice(), vm.clone()).unwrap();
         restored_state = snapshot.read_section::<Test>("test").unwrap();
 
-        // The semantic fn must not be called.
+        // The 4 semantic fns must not be called at version 4.
         assert_eq!(restored_state.field0, 0);
         assert_eq!(restored_state.field4, vec![4, 3, 2, 1]);
     }
 
+    #[test]
+    fn test_write_section() {
+        let mut vm = VersionMap::new();
+        vm.new_version()
+            .set_type_version(Test::type_id(), 2)
+            .new_version()
+            .set_type_version(Test::type_id(), 3)
+            .new_version()
+            .set_type_version(Test::type_id(), 4);
+        let state = Test {
+            field_x: 0,
+            field0: 0,
+            field1: 1,
+            field2: 2,
+            field3: "test".to_owned(),
+            field4: vec![4, 3, 2, 1],
+        };
+
+        // Serialize as v1.
+        let mut snapshot = Snapshot::new(vm.clone(), 1);
+        let mut section_len = snapshot.write_section("test", &state).unwrap();
+        // 8 bytes (field_x) + 8 bytes (field0) + 4 bytes (field1).
+        assert_eq!(section_len, 20);
```

Comment:
I think we do not need to check for the serialized length directly. This comes for free by validating A == B where A=deserialize(C), C=serialize(struct), B=deserialize(harcoded_blob). The hardcoded blob contains C when the test is created.

Reply:
This test was more of a sanity check, but I agree that `write_section()` is tested in other tests behind the scenes. I can remove this test.

Reply:
Done.